### PR TITLE
feat: allow duplicate availability

### LINKED ISSUE #9528: Add duplicate button to availability
### Is your proposal related to a problem?

The availability page right now has only delete option, it would be great to have duplicate one as well. 
image

https://app.cal.com/availability

This will be helpful to create a schedule with minor changes. 

### Describe the solution you'd like
add option to duplicate. 

(Describe your proposed solution here.)

### Describe alternatives you've considered

(Write your answer here.)

### Additional context

(Write your answer here.)

### PR DESCRIPTION
**Pull request by iamr-kumar**
## What does this PR do?

Fixes #9528 

## Demo

https://www.loom.com/share/c4fd5f14da8b40ba93d58ed32375cc72?sid=57435832-7b7b-4f8e-846b-2e3a69526471

## Type of change

- New feature (non-breaking change which adds functionality)

## How should this be tested?

- In the availability page, duplicate an existing availability and see if it's getting duplicated.

## Mandatory Tasks
- [x] Make sure you have self-reviewed the code. A decent size PR without self-review might be rejected.

### CODE REVIEW

**Comment by iamr-kumar** [comment]
@CarinaWolli  Hi, need some help on this feature. Couldn't get a conversation going on the issue page so implemented an initial version based on my understanding of what the requirement is. Would love to get some thoughts here.

**Comment by iamr-kumar** [comment]
If we go ahead with this PR, then a translation string would also need to be added, something like `availability duplicated successfully`.

**Comment by iamr-kumar** [comment]
> If we go ahead with this PR, then a translation string would also need to be added, something like `availability duplicated successfully`.

My bad, this is not needed. We can just use `availability created` message as we are technically creating a new schedule.

File: packages/trpc/server/routers/viewer/availability/schedule/duplicate.handler.ts [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+          id: user.id,
+        },
+      },
+      timeZone: schedule.timeZone ?? user.timeZone,
+      availability: {
+        createMany: {
+          data: availability.map((schedule) => ({
+            days: schedule.days,
+            startTime: schedule.startTime,
+            endTime: schedule.endTime,
```
Comment (CarinaWolli) [inline packages/trpc/server/routers/viewer/availability/schedule/duplicate.handler.ts:59]:
```suggestion
            endTime: schedule.endTime,
            date: schedule.date
```

**Review by CarinaWolli — DISMISSED** [review]
Sorry, I totally missed that you tagged me here. But it looks like you were doing a great job anyways 👏🏻

I think the basic version without the modal is perfectly fine 👍 Everything looks good to me, just one small bug that date overrides are not duplicated (see comment)

File: packages/trpc/server/routers/viewer/availability/schedule/duplicate.handler.ts (hunk 1 above)
Comment (iamr-kumar) [reply packages/trpc/server/routers/viewer/availability/schedule/duplicate.handler.ts:54]: Thanks for the suggestion. I have added the same. I have also removed getting the `eventType` while fetching the schedule as it is not really required for duplicating, thus avoiding additional join operation.

**Comment by iamr-kumar** [comment]
Hi @CarinaWolli, could you take a look at this?

**Review by CarinaWolli — APPROVED** [review]
Great work 👏🏻 Thank you for your contribution :)

**Comment by iamr-kumar** [comment]
Hey @CarinaWolli, just wanted to know if you are looking to merge this PR.

**Pull request merged**